### DISCUSSION TITLE
Dagger no resolving /etc/hosts

### DISCUSSION THREAD

**Discussion by dmarteau** [dagger/dagger#12477] [Help]
### What happened? What did you expect to happen?

Using private registry with private CA (stored in `/etc/docker/certs.d`, we store the ip resolution in /etc/hosts.  

It appears that Dagger do not take into account the /etc/hosts and fail to resolve the registry fqdn:
```
dagger.exceptions.QueryError: failed to do request: Head "...": dial tcp: lookup <fqdn> on 10.87.0.1:53: no such host
```

Is there a way to tell Dagger to resolve the registry using the /etc/hosts and use the certificats for the local docker settings ?

**Comment by helderco** [comment]
See:
- https://github.com/dagger/dagger/issues/4217
